packages/presentation/src/getIntentState.ts

```diff
@@ -1,44 +1,60 @@
 import {uuid} from '@sanity/uuid'
-import {type SearchParam} from 'sanity/router'
+import {type RouterState, type SearchParam} from 'sanity/router'
 
 import {encodeJsonParams, getPublishedId} from './internals'
-import type {PresentationStateParams} from './types'
+import type {PresentationSearchParams, PresentationStateParams} from './types'
+
+const preservedSearchParamKeys: Array<keyof PresentationSearchParams> = [
+  'preview',
+  'perspective',
+  'viewport',
+]
 
 /**
  * @internal
  */
 export function getIntentState(
   intent: string,
   params: Record<string, string>,
-  _routerState: undefined,
+  routerState: RouterState | undefined,
   payload: unknown,
 ):
   | (PresentationStateParams & {_searchParams: SearchParam[]})
   | {intent: string; params: Record<string, string>; payload: unknown} {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const {id, mode, path, presentation, type, ...searchParams} = params
 
+  const preservedSearchParams = (routerState?._searchParams || [])
+    // @todo Casting https://github.com/microsoft/TypeScript/issues/14520
```

Comment:
We can probably remove the casting after we upgrade to TS 5.5 😄

Reply:
Great! 🥳